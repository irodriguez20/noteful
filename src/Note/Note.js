import React from "react";
import { Link } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import ApiContext from "../ApiContext";
import config from "../config";
import "./Note.css";
import { format } from "date-fns";
//import PropTypes from "prop-types";

export default class Note extends React.Component {
  static contextType = ApiContext;

  handleClickDelete = e => {
    e.preventDefault();
    const noteId = this.props.id;
    // this.context.deleteNote(noteId);

    fetch(`${config.API_ENDPOINT}/notes/${noteId}`, {
      method: "DELETE",
      headers: {
        "content-type": "application/json"
      },
    })
      .then(res => {
        console.log('Delete res', res)
        if (!res.ok)
          return res.json().then(e => Promise.reject(e));
        return res.json();
      })
      .then(() => {
        this.context.deleteNote(noteId);
        this.props.onDeleteNote(noteId);
        this.props.history.push('/')
      })

      .catch(error => {
        console.error({ error });
      });
  };

  render() {
    const { name, id, modified } = this.props;
    return (
      <div className="Note">
        <h2 className="Note__title">
          <Link to={`/note/${id}`}>{name}</Link>
        </h2>
        <button
          className="Note__delete"
          type="button"
          onClick={this.handleClickDelete}
        >
          <FontAwesomeIcon icon="trash-alt" />
          {" "}
          remove
        </button>
        <div className="Note_dates">
          <div className="Note_dates-modified">
            Modified{" "}
            <span className="Date">
              {modified ? format(new Date(Date.parse(modified)), "yyyy-MM-dd") : ""}
            </span>
          </div>
        </div>
      </div>
    );
  }
}

