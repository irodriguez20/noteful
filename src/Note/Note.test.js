import React from "react";
import Note from "./Note";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe(`Note component `, () => {
  const props = {
    id: "a",
    name: "test-class-name"
  };

  it("renders a .Note by default", () => {
    const wrapper = shallow(<Note />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the Note given props", () => {
    const wrapper = shallow(<Note {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
