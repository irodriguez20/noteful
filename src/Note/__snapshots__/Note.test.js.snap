// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component  renders a .Note by default 1`] = `
<div
  className="Note"
>
  <h2
    className="Note__title"
  >
    <Link
      to="/note/undefined"
    />
  </h2>
  <button
    className="Note__delete"
    onClick={[Function]}
    type="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="trash-alt"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
     
    remove
  </button>
</div>
`;

exports[`Note component  renders the Note given props 1`] = `
<div
  className="Note"
>
  <h2
    className="Note__title"
  >
    <Link
      to="/note/a"
    >
      test-class-name
    </Link>
  </h2>
  <button
    className="Note__delete"
    onClick={[Function]}
    type="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="trash-alt"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
     remove
  </button>
  <div
    className="Note_dates"
  >
    <div
      className="Note_dates-modified"
    >
      Modified
       
      <span
        className="Date"
      >
        2019-01-15
      </span>
    </div>
  </div>
</div>
`;
